scalafix-diff/shared/src/main/scala/scalafix/diff/GitDiffParser.scala

```diff
@@ -0,0 +1,193 @@
+package scalafix.diff
+
+import scala.util.matching.Regex
+
+import java.nio.file.Path
+
+case class Range(start: Int, length: Int)
+
+sealed trait GitDiff
```

Comment:
Can we introduce a `case class GitDiffs(diffs: List[GitDiff])` instead of passing around `List[GitDiff]` everywhere?

I think we can then move `GitDiffParser` to `scalafix.internal.diff` and add a constructor via
```scala
object GitDiffs {
  def apply(input: Iterator[String], workingDir: Path): GitDiffs =
    new scalafix.internal.diff.GitDiffParser(input, workingDir).parse()
}
```
That way we minimize the surface of our public API.